update/v1alpha1/types_update_status_progress.go

```diff
@@ -0,0 +1,424 @@
+package v1alpha1
+
+import metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+// ClusterVersionStatusInsight reports the state of a ClusterVersion resource (which represents a control plane
+// update in standalone clusters), during the update.
+type ClusterVersionStatusInsight struct {
+	// conditions provides detailed observed conditions about ClusterVersion
+	// +listType=map
+	// +listMapKey=type
+	// +patchStrategy=merge
+	// +patchMergeKey=type
+	// +optional
+	// +kubebuilder:validation:MaxItems=10
+	Conditions []metav1.Condition `json:"conditions,omitempty" patchStrategy:"merge" patchMergeKey:"type"`
+
+	// resource is the ClusterVersion resource that represents the control plane
+	//
+	// Note: By OpenShift API conventions, in isolation this should be a specialized reference that refers just to
+	// resource name (because the rest is implied by status insight type). However, because we use resource references in
+	// many places and this API is intended to be consumed by clients, not produced, consistency seems to be more valuable
+	// than type safety for producers.
+	// +required
+	Resource ResourceRef `json:"resource"`
+
+	// assessment is the assessment of the control plane update process
+	// +required
+	Assessment ControlPlaneAssessment `json:"assessment"`
+
+	// versions contains the original and target versions of the upgrade
+	// +required
+	Versions ControlPlaneUpdateVersions `json:"versions"`
+
+	// completion is a percentage of the update completion (0-100)
```

Comment:
It surprises me (still) that we can calculate a completion percentage, how will this actually work?

Reply:
So right now this corresponds to what web console shows for control plane. 100% == all cluster operators updated. There are ~30 cluster operators so you basically jump by 3% iterations as they update. It is not linear but it is better than nothing. People like numbers going up.

It is intentionally not defined what the completion actually means - the informer is the authority and its "I'm 60% done" is just the platform expressing some opinion about itself. If we find better ways to measure completeness we can simply start using that in the backend.